### DISCUSSION TITLE
Can't pair with BlueRetro.io and PS4 controller

### DISCUSSION THREAD

**Discussion by luisstecca** [darthcloud/BlueRetro#682] [Q&A] [answered]
I have some ESP32 with this issue.
Wii controller and Switch pair normally, but PS4 controller does not. Also, I can't connect to BlueRetro.io to map the buttons.

P.S.: There are no other controllers paired with the ESP32 or other devices paired with my phone.

(Sorry for my english...)

**Comment by luisstecca** [comment]
Apparently, ESP32-D0WD-V3 models do not fully work for BlueRetro.

**Reply by darthcloud** [reply, marked as answer]
ESP32-D0WD-V3 / ESP32-D0WDR2-V3 (within a ESP-WROOM-32E module) is precisely the only thing supported.

~~Your problem sound like you got a rogue BT device connecting to BlueRetro.~~

~~Like a BLE LED strip or a BLE audio device.~~

Edit: missed the part you said Wii connect fine.

It's possible all the ESP-WROOM-32E I had were all R2 variant.

**Reply by luisstecca** [reply]
So that might be it.
I will take care to buy only the functional versions. Thanks for the reply and thanks a lot for the project. The retro world thanks you.

**Comment by necrosaromx** [comment]
Hi

I flashed a ESP32-D0WDQ successfully and Worked like a charm with FW v1.9.1_hw1, but I'm facing problems with an ESP32-D0WD-V3 variant while trying to pair the following controllers (the ones that I have for testing)

- Wii U Pro Controller
- XBOX Series S/X Controller
- 8Bitdo NeoGeo Controller
- Switch JoyCon

None of these controllers pair, status led keeps flashing and logs show no details (attached file)

[ESP32-D0WD-V3.txt](https://github.com/darthcloud/BlueRetro/files/14251089/ESP32-D0WD-V3.txt)

Are there special instructions to flash ESP32-D0WD-V3 variants?

Thanks
